oak-core-spi/src/main/java/org/apache/jackrabbit/oak/spi/mount/SimpleMountInfoProvider.java

```diff
@@ -67,9 +70,9 @@ public boolean hasNonDefaultMounts() {
     }
 
     @Override
-    public Collection<Mount> getMountsPlacedUnder(String path) {
-        Collection<Mount> mounts = new ArrayList<>();
-        for ( Mount mount : this.mounts.values()) {
+    public @NotNull Collection<Mount> getMountsPlacedUnder(String path) {
+        Collection<Mount> mounts = new ArrayList<>(this.mounts.size());
```

Comment:
If I'm not mistaken, the average size of the result is 1, so actually `new ArrayList<>(1);` would be better.
```suggestion
        Collection<Mount> mounts = new ArrayList<>(1);
```